### ISSUE TITLE
When calling detector(scan), the code is blocked

### ISSUE THREAD

**Issue by 2068143233** [VisualComputingInstitute/2D_lidar_person_detection#16]
When I'm using the test code in the readme, here's the code:：
```
import numpy as np
import sys
sys.path.append("/home/orin/2D_lidar_person_detection/src/dr_spaam")
from dr_spaam.detector import Detector

ckpt = 'check_point/self_supervised_person_detection/ckpt_jrdb_ann_ft_dr_spaam_e20.pth'
# print("ksajdfssas")
detector = Detector(
    ckpt,
    model="DR-SPAAM",          # Or DR-SPAAM
    gpu=True,               # Use GPU
    stride=1,               # Optionally downsample scan for faster inference
    panoramic_scan=True     # Set to True if the scan covers 360 degree
)
# print("ksajdf")
# tell the detector field of view of the LiDAR
laser_fov_deg = 360
# print("1")
detector.set_laser_fov(laser_fov_deg)
# print("2")
# detection
num_pts = 1090
# print("3")
while True:
    # create a random scan
    scan = np.random.rand(num_pts)  # (N,)
    # print("4")
    # detect person
    dets_xy, dets_cls, instance_mask = detector(scan)  # (M, 2), (M,), (N,)
    # print("5")
    print(detector(scan))
    # confidence threshold
    cls_thresh = 0.5
    cls_mask = dets_cls > cls_thresh
    dets_xy = dets_xy[cls_mask]
    dets_cls = dets_cls[cls_mask]
```
My program is blocked on line 29, the program doesn't report errors, but it doesn't continue to run, I don't know why, my platform is Jetson Orin, then python=3.8.10, pytorch=1.14,Eventually, I determined that the direct blocking occurred at about 100 lines of /dr_spaam/model/dr_spaam.py at
` 
out = self._conv_and_pool(out, self.conv_block_1) # /2 
`
by print

**Comment by 2068143233** [comment]
@Pandoro

**Comment by Pandoro** [comment]
I'm sorry, but this is impossible for us to debug. We do not have such a platform at hand and you are pretty much just telling me that the code hangs somewhere. The `_conv_and_pool` function is super simple and I don't see why this should hang.

Are you sure it actually hangs or is it just running extremely slow? Given the Orin is a more recent device and this code is a bit older, maybe there are some incompatibilities. I recommend finding the exact function call the method hangs in, make sure it actually hangs and isn't just slow and then try to figure out this happens, maybe with a toy example running on the same device.

**Comment by 2068143233** [comment]
> I'm sorry, but this is impossible for us to debug. We do not have such a platform at hand and you are pretty much just telling me that the code hangs somewhere. The `_conv_and_pool` function is super simple and I don't see why this should hang.
> 
> Are you sure it actually hangs or is it just running extremely slow? Given the Orin is a more recent device and this code is a bit older, maybe there are some incompatibilities. I recommend finding the exact function call the method hangs in, make sure it actually hangs and isn't just slow and then try to figure out this happens, maybe with a toy example running on the same device.

Thank you very much for your answer, I seem to have found the problem, it is a configuration problem with my cuda, which causes the GPU resources to be unable to be called, and it can be run by changing it to CPU, but can I directly use the point cloud bag collected by 2D lidar to test the dr_spaam_ros.launch, or what data format requirements need to be met to successfully use the model, I am using sensor_msgs/ LaserScan type, doesn't seem to work

**Comment by Pandoro** [comment]
Given what I can quickly gather from the ros node, this is specifically tailored towards the rosbags from the JRDB dataset. You'll have to check the code and modify it to work with another bag. I'm assuming this should be fairly straightforward though.

Given that the issue seems to be with your hardware and I can't offer support to modify this code to work on other rosbags, I'm closing this issue for now. Feel free to re-open / open another issue if you run into other problems.

**Comment by 2068143233** [comment]
> Given what I can quickly gather from the ros node, this is specifically tailored towards the rosbags from the JRDB dataset. You'll have to check the code and modify it to work with another bag. I'm assuming this should be fairly straightforward though.
> 
> Given that the issue seems to be with your hardware and I can't offer support to modify this code to work on other rosbags, I'm closing this issue for now. Feel free to re-open / open another issue if you run into other problems.

alright, thank you very much
